core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -67,6 +71,16 @@ object Partitioner {
       }
     }
   }
+
+  /**
+   * Returns true if the number of partitions of the RDD is either greater than or is
+   * less than and within a single order of magnitude of the max number of upstream partitions;
+   * otherwise, returns false
+   */
+  private def isEligiblePartitioner(hasMaxPartitioner: RDD[_], rdds: Seq[RDD[_]]): Boolean = {
+    val maxPartitions = rdds.map(_.partitions.length).max
+    log10(maxPartitions).floor - log10(hasMaxPartitioner.getNumPartitions).floor < 1
```

Comment:
Why `.floor` ?
It causes unnecessary discontinuity imo, for example: (9, 11) will not satisfy - but it should.

Reply:
Hi @mridulm , I suppose I was trying to ensure a strict order-of-magnitude check; but, I agree it leads to a discontinuity. I will change this, and the corresponding test cases.